### PR TITLE
Improve error handling in CLI

### PR DESCRIPTION
**Pull request by matthieucan**
Improve error handling:
- If `dbt parse` fails to run, a nice(r) error message is displayed
- In case of any error, the return code is 2
- Logging is written to stderr, instead of stdout

### CODE REVIEW

File: src/dbt_score/cli.py [hunk 1]
```diff
@@ -138,6 +138,14 @@ def lint(
         )
         ctx.exit(2)
 
+    except DbtParseException:
+        logger.error("dbt failed to parse project. Run `dbt parse` to investigate.")
+        ctx.exit(2)
+
+    except Exception:
+        logger.error(traceback.format_exc())
+        ctx.exit(2)
```
Comment (druzhinin-kirill) [inline src/dbt_score/cli.py:147]: I think it might not be 100% correct:

> Exit code 2 signifies invalid usage of some shell built-in command.

While here, the exception might be propagated from `lint_dbt_project` and relate to any internal bug. 
I think we can avoid broad exception handling and allow it propagate as is.

Comment (matthieucan) [reply src/dbt_score/cli.py:147]: I want to distinguish those 3 situations:
- Linter ran, and project is ok (warnings below threshold)
- Linter ran, and project is not ok (warnings above threshold)
- Unexpected error

The first one should be 0. I'm open to any exit code for the second and third situations, but they should be different. This approach is quite common, see e.g. black: https://black.readthedocs.io/en/stable/usage_and_configuration/the_basics.html#check

Would you suggest something else than 1 and 2?

Comment (druzhinin-kirill) [reply src/dbt_score/cli.py:147]: I believe broad exception handling should result in exit code 1. Exit code 2 is a bit misleading IMO as it identifies an invalid command call (while here a command is valid).
We build on top of `click` that follows this and has 2 exit code for [UsageError](https://github.com/pallets/click/blob/123dd717439d8620d8d6be5574d2c9f007952326/click/exceptions.py#L43). So 2 is sort of "reserved" for a specific meaning.

However, I checked other tools and it's not a strict standard and some tools have their own exit code interpretation. Also, checked a good explanation for [Ruff](https://docs.astral.sh/ruff/linter/#exit-codes) exit codes. 
So indeed here, they use 2 for any kind of errors (like you implemented initially) 👌 

TL/DR everything works if we document it 😄

**Review by druzhinin-kirill — APPROVED** [review]

File: src/dbt_score/cli.py (hunk 1 above)
Comment (matthieucan) [reply src/dbt_score/cli.py:147]: Indeed, POSIX only specifies 0 for success and >0 for failure.
On top of this, it's only conventions and/or specific decisions by programs.
Glad to hear Ruff follows the same, I'm pretty sure I've seen this in other linters too, but we can adapt this convention as well, and revise later if we find it causes issues.

I'll document it!

Comment (matthieucan) [reply src/dbt_score/cli.py:147]: Done, PTAL :)

**Comment by jochemvandooren** [comment]
Nice :rocket:
